### PR TITLE
PerformanceResourceTiming: RedirectEnd

### LINKED ISSUE #21257: PerformanceResourceTiming: RedirectEnd
In components/net/http_loader.rs::http_fetch we need to set redirect_end.

On getting, the redirectEnd attribute MUST return as follows:

1. The time immediately after receiving the last byte of the response of the last redirect, if there are HTTP redirects or equivalent when fetching the resource and all the redirects or equivalent pass the timing allow check algorithm.
2. zero, otherwise.

https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-redirectend

### PR DESCRIPTION
**Pull request by verma-varsha**
---
 
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21257  (GitHub issue number if applicable)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22561)

### CODE REVIEW

**Comment by verma-varsha** [comment]
@jdm @avadacatavra Can you please help me get the changes reviewed?

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -644,7 +644,12 @@ pub fn http_fetch(
         };
     }
 
-    // TODO redirect_end: last byte of response of last redirect
+    // redirect_end: last byte of response of last redirect
+    context
+        .timing
+        .lock()
+        .unwrap()
+        .set_attribute(ResourceAttribute::RedirectEnd);
```
Comment (jdm) [inline components/net/http_loader.rs]: Unfortunately I think the TODO was misleading. Much like my review comment in #22319, step 20 of https://w3c.github.io/resource-timing/#processing-model suggests that this should be set in a different part of the code, and will be very similar to how we implement responseEnd (#21263). This part of the code has only just received the HTTP headers, not the response body.

**Review by jdm — CHANGES_REQUESTED** [review]
The changes in this PR implement the timing part, but we will also need to implement the DOM API in components/script/dom/webidls/PerformanceResourceTiming.webidl and components/script/dom/performanceresourcetiming.rs.

**Comment by jdm** [comment]
@verma-varsha Are you still planning to complete this work?

**Comment by verma-varsha** [comment]
@jdm I was caught up with some other work which led me to losing track of this. Thanks for reminding. I'll try to update the PR in a few days.

**Comment by jdm** [comment]
I'm going to go ahead and close this due to inactivity. If you return to it, go ahead and reopen it!

**Pull request closed without merging**